src/spyglass/position/position_merge.py

```diff
@@ -50,163 +40,32 @@ class DLCPosV1(dj.Part):
-        if position_id is None:
-            key["position_id"] = (
-                dj.U().aggr(self & key, n="max(position_id)").fetch1("n") or 0
-            ) + 1
-        else:
-            id = (self & key).fetch("position_id")
-            if len(id) > 0:
-                position_id = max(id) + 1
-            else:
-                position_id = max(0, position_id)
-            key["position_id"] = position_id
-        super().insert1(key, **kwargs)
-        source = key["source"]
-        if source in ["Common"]:
-            table_name = f"{source}Pos"
-        else:
-            version = key["version"]
-            table_name = f"{source}PosV{version}"
-        part_table = getattr(self, table_name)
-        # TODO: The parent table to refer to is hard-coded here, expecting it to be the second
-        # Table in the definition. This could be more flexible.
-        if params:
-            table_query = (
-                dj.FreeTable(dj.conn(), full_table_name=part_table.parents()[1])
-                & key
-                & params
-            )
-        else:
-            table_query = (
-                dj.FreeTable(dj.conn(), full_table_name=part_table.parents()[1])
-                & key
-            )
-        if any(
-            "head" in col
-            for col in list(table_query.fetch().dtype.fields.keys())
-        ):
-            (
-                analysis_file_name,
-                position_object_id,
-                orientation_object_id,
-                velocity_object_id,
-            ) = table_query.fetch1(
-                "analysis_file_name",
-                "head_position_object_id",
-                "head_orientation_object_id",
-                "head_velocity_object_id",
-            )
-        else:
-            (
-                analysis_file_name,
-                position_object_id,
-                orientation_object_id,
-                velocity_object_id,
-            ) = table_query.fetch1(
-                "analysis_file_name",
-                "position_object_id",
-                "orientation_object_id",
-                "velocity_object_id",
-            )
-        part_table.insert1(
-            {
-                **key,
-                "analysis_file_name": analysis_file_name,
-                "position_object_id": position_object_id,
-                "orientation_object_id": orientation_object_id,
-                "velocity_object_id": velocity_object_id,
-                **params,
-            },
-        )
-
-    def fetch_nwb(self, *attrs, **kwargs):
-        source = self.fetch1("source")
-        if source in ["Common"]:
-            table_name = f"{source}Pos"
-        else:
-            version = self.fetch1("version")
-            table_name = f"{source}PosV{version}"
-        part_table = getattr(self, table_name) & self
-        return part_table.fetch_nwb()
-
     def fetch1_dataframe(self):
         nwb_data = self.fetch_nwb()[0]
```

Comment:
@CBroz1 is this syntax correct? Or should we extract the restriction included in self and change this to: (?)
``` python
restriction = ...
nwb_data = self.fetch_nwb(restriction)
```
Another option would be to require the restriction be passed as an arg instead of the current use:
``` python
# current way to call fetch1_dataframe
(merge_table & restriction).fetch1_dataframe()
# potential new way
merge_table.fetch1_dataframe(restriction)
```
Although that approach would differ from the rest of the code base.

Reply:
Looks like 
``` python
nwb_data = self.fetch_nwb(self.proj())[0]
```
Thoughts?